src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,357 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   Tweaked colour picker allows for tooltip texts translations and has a customized theming.
+/// </summary>
+public class TweakedColourPicker : ColorPicker
+{
+    /// <summary>
+    ///   This is where presets are stored after a colour picker exited scene tree.
+    /// <remarks>
+    ///   KeyValuePair&lt;GetPath(), GetPresets()&gt;:
+    ///   <ul>
+    ///     <li> First parameter (string) is the string format of NodePath; </li>
+    ///     <li> Second parameter (Color[]) is the preset colours to be stored. </li>
+    ///   </ul>
+    /// </remarks>
+    /// </summary>
+    private static readonly List<KeyValuePair<string, Color[]>> PresetsStorage
+        = new List<KeyValuePair<string, Color[]>>();
+
+    private readonly List<TweakedColourPickerPreset> presets = new List<TweakedColourPickerPreset>();
+
+    private HSlider sliderROrH;
+    private HSlider sliderGOrS;
+    private HSlider sliderBOrV;
+    private HSlider sliderA;
+    private ToolButton pickerButton;
+    private CheckButton hsvCheckButton;
+    private CheckButton rawCheckButton;
+    private LineEdit htmlColourEdit;
+    private HSeparator separator;
+    private GridContainer presetsContainer;
+    private TextureButton addPresetButton;
+
+    private bool hsvButtonEnabled = true;
+    private bool rawButtonEnabled = true;
+    private bool presetsEnabled = true;
+    private bool presetsVisible = true;
+
+    /// <summary>
+    ///   Decide if user can toggle HSV CheckButton to switch HSV mode.
+    /// </summary>
+    [Export]
+    public bool HSVButtonEnabled
+    {
+        get => hsvButtonEnabled;
+        set
+        {
+            hsvButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    /// <summary>
+    ///   Decide if user can toggle Raw CheckButton to switch Raw mode.
+    /// </summary>
+    [Export]
+    public bool RawButtonEnabled
+    {
+        get => rawButtonEnabled;
+        set
+        {
+            rawButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool HsvMode
+    {
+        get => base.HsvMode;
+        set
+        {
+            if (value && RawMode)
+                return;
+
+            base.HsvMode = value;
+            UpdateButtonsState();
+
+            // Update HSV sliders' tooltips
+            UpdateTooltips();
+        }
+    }
+
+    [Export]
+    public new bool RawMode
+    {
+        get => base.RawMode;
+        set
+        {
+            if (value && HsvMode)
+                return;
+
+            base.RawMode = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool PresetsEnabled
```

Comment:
So this is about being able to add new presets? As there's separately a presets visible property. If so I think this should maybe be named `PresetsEditable`

Reply:
This also overrides a Godot property. Can't rename it because we can't rename Godot's properties or disable them.

At most we an add a Obsolete Attribute and rename the functional one to our own name, but we still need to override this.

Reply:
I think it's fine to keep then. Could add a comment explaining with remark why this is named like this.